src/components/HeaderTodo/HeaderTodo.tsx

```diff
@@ -0,0 +1,109 @@
+import classNames from 'classnames';
+import { FC, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import React from 'react';
+import { errorNotification } from '../../constants/errors';
+import { addTodos, USER_ID } from '../../api/todos';
+import { focusTodoInput } from '../../helpers/inputFocus';
+
+interface Props {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: (message: string) => void;
+  setIsTempTodoCreating: (isCreating: boolean) => void;
+  setTempTodo: (todo: Todo | null) => void;
+  isTempTodoCreating: boolean;
+  toggleAllTodos: () => void;
+}
+
+export const HeaderTodo: FC<Props> = ({
+  todos,
+  setErrorMessage,
+  setTodos,
+  setIsTempTodoCreating,
+  setTempTodo,
+  isTempTodoCreating,
+  toggleAllTodos,
```

Comment:
You have too many props
The component duplicates some functionality and becomes too complex

I would suggest refactor it to have fewer props:
```suggestion
  toggleAllTodos: () => void,
  onAddTodo: (todo: Todo) => void
```

And take away all complex todo-creating logic